### PR TITLE
Token Persistence

### PR DESCRIPTION
**Pull request by nreynis**
I retrofited Sevastian Hübner work (@disasterdrop) on Token Persistence.
Extracted the changes from his hard fork, added documentation and test.

I credited him in PhpDoc `@author` field.

### CODE REVIEW

**Comment by nreynis** [comment]
A nice addition to this would be a PSR-16 (simple cache) adapter. If you want I can add it to the PR.

**Comment by eljam** [comment]
Hello,

Yes, a few month i was thinking about adding doctrine cache library. But now that we have PSR-16 adapter it would be a good addition to the library.

if you don't mind, it would be great :)

File: Tests/Persistence/TokenPersistenceTest.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        $simpleCache = new FilesystemCache();
+        $tokenPersistence = new SimpleCacheTokenPersistence($simpleCache);
+        $token = new JwtToken('foo', new \DateTime('now'));
+
+        $tokenPersistence->saveToken($token);
+
+        $this->assertTrue($tokenPersistence->hasToken());
+        $this->assertEquals($tokenPersistence->restoreToken()->getToken(), $token->getToken());
+		
+		$tokenPersistence->deleteToken();
```
Comment (eljam) [inline Tests/Persistence/TokenPersistenceTest.php]: need be align a bit.

Comment (nreynis) [reply Tests/Persistence/TokenPersistenceTest.php]: Yep I was in a hurry last friday, I also forgot to stage the readme. I'll fix it tomorrow.

Comment (eljam) [reply Tests/Persistence/TokenPersistenceTest.php]: Ok cool. Thank you for your contribution !

Comment (nreynis) [reply Tests/Persistence/TokenPersistenceTest.php]: I made the edits and squashed the changes in a single commit for cleaner history.

Comment (eljam) [reply Tests/Persistence/TokenPersistenceTest.php]: Cool thx

Comment (nreynis) [reply Tests/Persistence/TokenPersistenceTest.php]: 🎉 Thank you !

**Pull request merged**